test/metabase/upload_test.clj

```diff
@@ -1022,89 +1026,87 @@
+                       :schema    (re-pattern (str "(?i)" schema-name))
+                       :is_upload true}
+                      new-table)
+                  "A new table is created")
+              (is (= "complete"
+                     (:initial_sync_status new-table))
+                  "The table is synced and marked as complete")
+              (is (= #{["_mb_row_id" :type/PK]
+                       ["id"   :type/PK]
+                       ["name" :type/Name]}
+                     (->> (t2/select Field :table_id (:id new-table))
+                          (map (fn [field] [(u/lower-case-en (:name field))
+                                            (:semantic_type field)]))
+                          set))
+                  "The sync actually runs")
+              (is (true? @in-future?)
+                  "Table has been synced in a separate thread"))))
+        (finally
+          (t2/update! :model/Database db-id original-sync-values))))))
 
 (deftest create-csv-upload!-table-prefix-test
   (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
-    (mt/with-empty-db
-      (testing "Happy path with table prefix, and without schema"
-        (if (driver/database-supports? driver/*driver* :schemas (mt/db))
-          (is (thrown-with-msg?
-                java.lang.Exception
-                #"^A schema has not been set."
-                (upload-example-csv! :table-prefix "uploaded_magic_" :schema-name nil)))
-          (let [new-model (upload-example-csv! :table-prefix "uploaded_magic_")
-                new-table (t2/select-one :model/Table (:table_id new-model))]
-            (is (=? {:name #"(?i)example csv file(.*)"}
-                    new-model))
-            (is (=? {:name #"(?i)uploaded_magic_example(.*)"}
-                    new-table))
-            (is (nil? (:schema new-table)))))))))
+    (testing "Happy path with table prefix, and without schema"
+      (if (driver/database-supports? driver/*driver* :schemas (mt/db))
+        (is (thrown-with-msg?
+              java.lang.Exception
+              #"^A schema has not been set."
+              (upload-example-csv! :table-prefix "uploaded_magic_" :schema-name nil)))
+        (with-upload-table!
+          [table (card->table (upload-example-csv! :table-prefix "uploaded_magic_"))]
+          (is (=? {:name #"(?i)example csv file(.*)"}
+                  (table->card table)))
+          (is (=? {:name #"(?i)uploaded_magic_example(.*)"}
+                  table))
+          (is (nil? (:schema table))))))))
 
 (deftest create-csv-upload!-auto-pk-column-display-name-test
   (testing "The auto-generated column display_name should be the same as its name"
    (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
-     (mt/with-empty-db
-       (let [new-model (upload-example-csv!)
-             new-field (t2/select-one Field :table_id (:table_id new-model) :name "_mb_row_id")]
+     (with-upload-table!
+       [table (card->table (upload-example-csv!))]
```

Comment:
nit: typically we keep bindings for `with` macros up on the first line.

Reply:
fixed for this case. d2344a9967c09a2bdb6e6407434fef58ea7e598e

I'm going to leave other cases such as [here](https://github.com/metabase/metabase/blob/d2344a9967c09a2bdb6e6407434fef58ea7e598e/test/metabase/upload_test.clj#L1046) where following this convention would make the code more unreadable to me.

There are many exceptions to this e.g. `mt/with-temp` where we add a new line after the macro symbol to avoid creeping to the right.

Reply:
Yeah those exceptions make sense